samza-core/src/main/java/org/apache/samza/standalone/PassthroughJobCoordinator.java

```diff
@@ -112,20 +110,10 @@ public void setListener(JobCoordinatorListener listener) {
   @Override
   public JobModel getJobModel() {
     JavaSystemConfig systemConfig = new JavaSystemConfig(this.config);
-    Map<String, SystemAdmin> systemAdmins = new HashMap<>();
-    for (String systemName: systemConfig.getSystemNames()) {
-      String systemFactoryClassName = systemConfig.getSystemFactory(systemName);
-      if (systemFactoryClassName == null) {
-        LOGGER.error(String.format("A stream uses system %s, which is missing from the configuration.", systemName));
-        throw new SamzaException(String.format("A stream uses system %s, which is missing from the configuration.", systemName));
-      }
-      SystemFactory systemFactory = Util.<SystemFactory>getObj(systemFactoryClassName);
-      systemAdmins.put(systemName, systemFactory.getAdmin(systemName, this.config));
-    }
-
-    StreamMetadataCache streamMetadataCache = new StreamMetadataCache(
-        Util.<String, SystemAdmin>javaMapAsScalaMap(systemAdmins), 5000, SystemClock.instance());
-
+    Map<String, SystemAdmin> systemAdminMap = systemConfig.getSystemAdmins();
```

Comment:
This is in a lot of places. Would it be cleaner to add a helper/constructor in SystemAdmins that does this?

Reply:
Not sure if I completely understand your comments. But I have updated the constructor of SystemAdmins to simplify the code. Can you see if it addresses the problem?

Reply:
Fixed as discussed.